src/main/java/seedu/parser/EditContactParser.java

```diff
@@ -28,8 +29,12 @@
      * @param detail         Unparsed detail
      * @throws InvalidFlagException If the flag given is not recognised
      */
-    public void parseDetail(String[] contactDetails, String detail) throws InvalidFlagException {
+    public void parseDetail(String[] contactDetails, String detail)
+            throws InvalidFlagException, MissingDetailException {
         String[] destructuredDetails = detail.split(" ", NUMBER_OF_EDIT_DETAILS);
+        if (destructuredDetails[1].trim().equalsIgnoreCase("")) {
+            throw new MissingDetailException();
+        }
```

Comment:
Hi @ng-andre , would like to check, will this be an IndexOutOfBoundsException? I am thinking from the perspective of the following input: `edit 1 -n -g ng-andre`

Following our discussion last night, would it be better to perhaps check for this by using the following code?
```
if (destructuredInputs.length < 2) {
    throw new MissingDetailException();
}
```
Do let me know!

Reply:
@marcusbory I have tested with the input `edit 2` (without spacing after 2) with the code you suggested, which would not enter the if block for some reason, which I suspect is due to the split function specifying 3. Not sure why this is the case, but it would usually not catch such an error, therefore I have changed it to this if block instead, which catches exceptions for both `edit 2` and `edit 2 ` (with extra space after 2). No IndexOutOfBoundsException was thrown during my testing to the best of my knowledge.